Sc updates in noise (#349)
aeternity/protocol: channels/OFF-CHAIN.md

### `ping`/`pong`

## Sub-messages

The following serializations can be consistently used below.

### Updates list

Progess off-chain is achieved by one participant proposing a set of updates to

> be applied on top of the last state as well with the next state that is

Reviewer: ```suggestion
be applied on top of the last state as well as the next state that is
```

produced.
A single off-chain update's serialization has the following structure:

>   name                  size (bytes)

Reviewer 2: Strange markup?

Once `funding_locked` messages have been exchanged, the channel enters the
`open` state. Changes to the off-chain state can be effected by sending an

> `update` message. It consists of a singly-signed off-chain transaction and

Reviewer: ```suggestion
`update` message. It consists of a single-signed off-chain transaction and
```

> an updates list that contains a list of operations to be performed on the

Reviewer: ```suggestion
a list of updates list containing the operations to be performed on the
```

previous state. The `state_hash` of the off-chain transaction is the
aggregated root hash of the resulting state trees. The receiver must verify

> that the state is a valid outcome, then return it, co-signed, in an

Reviewer: ```suggestion
that the state is the correct outcome, then return it, co-signed, in an
```

Message code: 11

In order to deposit more funds into the channel, one party can initiate

> a `deposit_created` request. It consists of singly signed `channel_deposit_tx`

Reviewer: ```suggestion
a `deposit_created` request. It consists of a single-signed `channel_deposit_tx`
```

Message code: 15

In order to withdraw coins from the channel, one party can initiate

> a `withdraw_created` request. It consists of a singly signed

Reviewer: ```suggestion
a `withdraw_created` request. It consists of a single-signed
```
